src/gpg/fn/print_key.py

```diff
@@ -0,0 +1,117 @@
+# Print the details of the given key id.
+import os
+import sys
+import textwrap
+import time
+
+import gpg
+
+
+def print_key(identity):
+    """
+    print the details of key whose identitiy is passed
+    """
+    c = gpg.Context()
+    key = list(c.keylist(identity, mode=gpg.constants.keylist.mode.SIGS))
+
+    # exit with status 1 if more than one key matches are found for identity
+    if len(key) > 1:
+        exit(1)
+    else:
+        key = key[0]
+
+    if os.environ["DEBUG"] == "yes":
+        print(key)
+        print()
+
+    # uid
+    uid_list = ["uid: " + user_id.uid for user_id in key.uids]
+    all_uids = "\n".join(uid_list)
+
+    # fpr
+    fpr = " ".join(textwrap.wrap(key.fpr, 4))
+
+    # trust
+    trust_map = {
+        "UNKNOWN": "UNKNOWN",
+        "UNDEFINED": "UNKNOWN",
+        "ULTIMATE": "ULTIMATE",
+        "NEVER": "NONE",
+        "MARGINAL": "MARGINAL",
+        "FULL": "FULL"
+        }
```

Comment:
I see only 4 levels of trust here: https://www.gnupg.org/gph/en/manual/x334.html
Can you specify the source?

Reply:
I used help(gpg.constants.validity) in the python shell to find all of these. You can find them in the c library also [here](url) highlighted below:
```c
/* The available validities for a trust item or key.  */
 311 typedef enum
 312   {
 313     GPGME_VALIDITY_UNKNOWN   = 0,
 314     GPGME_VALIDITY_UNDEFINED = 1,
 315     GPGME_VALIDITY_NEVER     = 2,
 316     GPGME_VALIDITY_MARGINAL  = 3,
 317     GPGME_VALIDITY_FULL      = 4,
 318     GPGME_VALIDITY_ULTIMATE  = 5
 319   }
 320 gpgme_validity_t;
```
I initially found these by looking at python tests though they don't contain all. you can look them up [here](https://git.gnupg.org/cgi-bin/gitweb.cgi?p=gpgme.git&a=search&h=HEAD&st=grep&s=constants.validity.)
I have mapped them to the ones listed above in your url and since undefined was extra, I mapped it to unknown only. UNDEFINED would have been treated as UNKNOWN in existing bash script also. This is also compatible with `egpg contact certify`.

Reply:
I understand that ULTIMATE is a trust level that you assign only to your own keys. However it is not clear to me the difference between UNKNOWN and UNDEFINED. Maybe UNDEFINED is the initial value (which means no-value), but you cannot assign UNDEFINED, you can only assign UNKNOWN.

This can be a good question to ask on gnupg-devel. But maybe this issue is not so important, after all.